ArduPlane/parachute.cpp

```diff
@@ -57,4 +58,37 @@ bool Plane::parachute_manual_release()
     return true;
 }
 
+/*
+    Parachute emergency check - trigger parachute when vertical speed threshold is reached
+    or a extreme low battery voltage threshold is reached.
+*/
+void Plane::parachute_emergency_check()
+{
+    // exit immediately if parachute or emergency mode is not enabled
+    if (!parachute.enabled() || !parachute.emergency_enabled() || parachute.released()) {
+        return;
+    }
+     
+    // do not release if vehicle is not flying
+    if (!is_flying()) {
+        return;
+    }
+     
+    // do not release if taking off 
+    if (auto_state.takeoff_complete == false ) {
+        return;
+    }
+    
+    // if the sink rate gets above the limits, release the parachute
+    if (auto_state.sink_rate > parachute.max_sink()) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL,"Emergency Parachute: Stall detected");
+        parachute_release();
+    }
+    else if (battery.voltage() < parachute.min_batt_voltage()) {
```

Comment:
This batt check needs hysteresis. Battery is notoriously noisy and fluctuates a lot depending on throttle and you don't want a single check to trigger the parachute. Perhaps what we need is a "super low-pass" version in the battery library because doing checks like are not the parachute's library duty to deal with.

Reply:
I will make a deep read on AP_BattMonitor

Reply:
I'll take care of the battery stuff right now

Reply:
Here's a PR https://github.com/ArduPilot/ardupilot/pull/9508

please give it a try on your branch

Reply:
Cool, i will do it